server/sonar-ce/src/main/java/org/sonar/ce/container/CePluginJarExploder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ce.container;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.commons.io.FileUtils;
+import org.picocontainer.Startable;
+import org.sonar.api.utils.TempFolder;
+import org.sonar.api.utils.ZipUtils;
+import org.sonar.core.platform.ExplodedPlugin;
+import org.sonar.core.platform.PluginInfo;
+import org.sonar.core.platform.PluginJarExploder;
+
+import static org.apache.commons.io.FileUtils.forceMkdir;
+
+/**
+ * Explode plugin JARs into a temporary directory
+ */
+public class CePluginJarExploder extends PluginJarExploder implements Startable {
+
+  private final TempFolder tempFolder;
+  private File rootDir;
+
+  public CePluginJarExploder(TempFolder tempFolder) {
+    this.tempFolder = tempFolder;
+  }
+
+  @Override
+  public void start() {
+    this.rootDir = tempFolder.newDir("ce-exploded-plugins");
+    try {
+      org.sonar.core.util.FileUtils.cleanDirectory(rootDir);
+    } catch (IOException e) {
+      throw new IllegalStateException("Can not clean " + rootDir, e);
+    }
+  }
+
+  @Override
+  public void stop() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1186)